src/main/java/seedu/address/model/module/Grade.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.model.module;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Module's grade in the transcript.
+ * <p>
+ * Guarantees: immutable; is valid as declared in {@link #isValidGrade(String)}
+ */
+public class Grade {
```

Comment:
Also for this, mind updating the lucidchart? 

Otherwise, looks good.

Reply:
Yep, going to update documentations for both lucidchart and the developers guide once i'm done adding test.